app/src/main/java/com/door43/translationstudio/newui/translate/TargetTranslationActivity.java

```diff
@@ -300,6 +349,240 @@ public boolean onMenuItemClick(MenuItem item) {
+
+                        }
+
+                        @Override
+                        public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+                        }
+
+                        @Override
+                        public void afterTextChanged(Editable s) {
+
+                            if(mSearchTimer != null) {
+                                mSearchTimer.cancel();
+                            }
+
+                            mSearchTimer = new Timer();
+                            mSearchTimerTask = new SearchTimerTask(TargetTranslationActivity.this, s);
+                            mSearchTimer.schedule(mSearchTimerTask, SEARCH_START_DELAY);
+                        }
+                    };
+                    edit.addTextChangedListener(mSearchTextWatcher);
+                } else {
+                    setSearchFilter(null); // clear search filter
+                }
+            }
+
+            ImageButton close = (ImageButton) searchPane.findViewById(R.id.close_search);
+            if(close != null) {
+
+                close.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        removeSearchBar();
+                    }
+                });
+            }
+
+            Spinner type = (Spinner) searchPane.findViewById(R.id.search_type);
+            if(type != null) {
+                List<String> types = new ArrayList<String>();
+                types.add(this.getResources().getString(R.string.search_source));
+                types.add(this.getResources().getString(R.string.search_translation));
+                ArrayAdapter<String> typesAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, types);
+                typesAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+                type.setAdapter(typesAdapter);
+
+                type.setOnItemSelectedListener(this);
+            }
+        }
+    }
+
+    /**
+     * sets the visible state of the search spinner
+     * @param displayed
+     */
+    private void setSearchSpinner(boolean displayed) {
+        if(mSearchingSpinner != null) {
+            mSearchingSpinner.setVisibility(displayed ?  View.VISIBLE : View.GONE);
+        }
+    }
+
+    /**
+     * called if search type is changed
+     * @param parent
+     * @param view
+     * @param pos
+     * @param id
+     */
+    public void onItemSelected(AdapterView<?> parent, View view,
+                               int pos, long id) {
+        setSearchFilter();
+    }
+
+    /**
+     * called if no search type is selected
+     * @param parent
+     */
+    public void onNothingSelected(AdapterView<?> parent) {
+        // do nothing
+    }
+
+    /**
+     * see if translation search is selected
+     */
+    private boolean isTranslationSearchSelected() {
```

Comment:
This is only used once and does not need to be a function.
We could perform this same action inline like 

```
boolean searchTarget = getSearchTypeSelection() == 1;
```